### PR TITLE
PP-29 Edit profile & form

### PR DESCRIPTION
**Pull request by vrudnitskaya**
### Description
**### !!!! You need to switch to the BACK branch named PP-47-lessons-model-router-controller-back
Please, before you start to check upload photos feature, be sure you have a folder on the back
Public(folder already exists)->Upload(folder to create inside public).
Otherwise you could get an error and will not be able to upload img to profile.**

The function of editing the user profile was implemented.
The user can separately edit information about themselves or simply change the profile image.
Once the information is changed, the user is redirected to the home page/profile page.

### Screenshots
The user's profile before the editing.
edit-profile-before

How the form looks like.
edit-profile-form

Some fields are required.
edit-profile-errors

edit-profile-filling

Information is automatically updated.
edit-profile-after

### **When the user clicks the Add button, a field to add a file is displayed. Then you need to click button "Upload"**

edit-profile-add-photo
edit-profile-add-photo2

The user is then redirected to the home page.
edit-profile-new_image

### CODE REVIEW

**Review by ofeterovskaya — APPROVED** [review]
I really love how the page looks like. You did great job for mobile version. 
The only 3nitpicks I would address are
1. Is it possible to remove vertical scrolling? 
2. The width of footer. On different devices it's too narrow sometimes
3. Could it be better to Say "Date of Birth" -- of-lowcase
 But I know it's so easy for to to fix, that's why I will approve this PR.
Everything looks good on my side

**Review by mariyador — APPROVED** [review]
Valya! Great job!
I can successfully update my profile.

**Comment by vrudnitskaya** [comment]
> I really love how the page looks like. You did great job for mobile version. The only 3nitpicks I would address are
> 
> 1. Is it possible to remove vertical scrolling?
> 2. The width of footer. On different devices it's too narrow sometimes
> 3. Could it be better to Say "Date of Birth" -- of-lowcase
>    But I know it's so easy for to to fix, that's why I will approve this PR.
>    Everything looks good on my side

Thank you, Oksana! I've fixed "Date of birth". I'm going to fix the scroll and the height of the footer in another ticket

File: src/components/pages/Dashboard.jsx
Comment (AnnaPestova1) [inline src/components/pages/Dashboard.jsx:17]: Why we do not handle error messages from BackEnd?

File: src/components/pages/Dashboard.jsx
Comment (AnnaPestova1) [inline src/components/pages/Dashboard.jsx:15]: do we need this console.log?

File: src/components/pages/dashboard-pages/EditProfile.jsx
Comment (AnnaPestova1) [inline src/components/pages/dashboard-pages/EditProfile.jsx:76]: I suppose we should handle API errors

File: src/components/pages/dashboard-pages/EditProfile.jsx
Comment (AnnaPestova1) [inline src/components/pages/dashboard-pages/EditProfile.jsx:45]: it is better to move large functions in a separate file

**Review by AnnaPestova1 — COMMENTED** [review]
Working as expected. But we cannot merge until the Backend changes are merged into dev branch. I also have some questions - see comments

File: src/components/pages/dashboard-pages/EditProfile.jsx
Comment (vrudnitskaya) [reply src/components/pages/dashboard-pages/EditProfile.jsx:45]: ok, I'll fix it

File: src/components/pages/dashboard-pages/EditProfile.jsx
Comment (vrudnitskaya) [reply src/components/pages/dashboard-pages/EditProfile.jsx:76]: ok,  I'll think how to this.

File: src/components/pages/dashboard-pages/EditProfile.jsx
Comment (JoHeyJo) [inline src/components/pages/dashboard-pages/EditProfile.jsx:36]: Let's follow common naming conventions and reserve the 'set' prefix for state setter functions.

File: src/components/pages/dashboard-pages/EditProfile.jsx
Comment (JoHeyJo) [inline src/components/pages/dashboard-pages/EditProfile.jsx:51]: Is `null` necessary or can we use the expected data type?

File: src/components/pages/dashboard-pages/HomeTeacher.jsx
Comment (JoHeyJo) [inline src/components/pages/dashboard-pages/HomeTeacher.jsx:34]: sp: specialty

**Review by JoHeyJo — CHANGES_REQUESTED** [review]
App is looking great! Some small things to change.

**Comment by vrudnitskaya** [comment]
> App is looking great! Some small things to change.
Thank you!
I'm going to make changes this evening.

**Comment by vrudnitskaya** [comment]
> App is looking great! Some small things to change.

Thank you!
I'm going to make changes this evening.

**Review by JoHeyJo — APPROVED** [review]
Looks good on my end. I don't see changes to some of Anna's comments. If the request is not possible leave a response explaining why. Great job, Valentina!

**Review by AnnaPestova1 — APPROVED** [review]
Hi Valentina! Thank you for changes!
I noticed an issue regarding error message (it is not disappear even if functionality is restored).  See screen record. Could you please to create ticket to handle this issue.

https://github.com/user-attachments/assets/d65dfe0e-1c0e-4e84-9c99-40ca44b75c3c


Thank you.
Overall everything looks great!

**Comment by vrudnitskaya** [comment]
> Hi Valentina! Thank you for changes! I noticed an issue regarding error message (it is not disappear even if functionality is restored). See screen record. Could you please to create ticket to handle this issue.
> 
>  Error.message.issue2.mov 
> Thank you. Overall everything looks great!

Anna, thanks for noticing this issue. I found out how to handle it. I will fix it and then merge the branch. No need to create a new ticket.

**Pull request merged**
